docs/index.md

```diff
@@ -116,15 +116,45 @@ after execute_perform
 => true
 ```
 
-### Performer
+### Context & performer
 
-Every BA has a performer which can be assigned via a `.as` class method before BA creation.
+Every BA has a context which is a hash that can be assigned via `.using` class method before BA 
+initialization. Context is usually used to pass performer of the action which is so common that there
+are methods defined to access & set performer
+`performer` specifically.
 
 ```ruby
-MyAction.as(Admin.first).new(params)
+action = MyAction.using(performer: Admin.first).new(params)
```

Comment:
@andriusch We're overwriting Ruby Core's `Module.using` ([doc](https://ruby-doc.org/core/Module.html#method-i-using)), can't we overload Granite's `as` instead?

Reply:
Sorry I still missed your comment. I'll update granite in another PR, you're right we shouldn't update `Module.using`, but I don't think it's a good idea to allow both `as(performer)` and `as(performer: performer)`. Maybe you have alternative name suggestions?

Reply:
@andriusch Why not?

`.as` has a good name semantic "execute this action **as** this role".
`.using` is generic and less specific.

If the goal is to introduce _keyword args_ for performer, can't we just change `.as` [signature](https://github.com/toptal/granite/blob/68095d6d4f136fdd82c4c2592f0a9a8eedee086b/lib/granite/context_proxy.rb#L17)?

```ruby
def as(perf = nil, performer: nil)
  actor = perf || performer
  # ...
end
```

This is backward compatible.

Reply:
@jodosha It's `using(context)` now. The concept is bigger and more generic than just performer.

Alternative name could be `with(context)`, `within(context)`, `inside(context)`. Although I like `using` the most so far.

Reply:
I think of all the alternatives I like `with(context)` the most.

Reply:
@paneq @andriusch OK, the PR description and most of the changes aren't advertising for context, just performer. 🤷 

If that is the case, `with(context)` LGTM.

Reply:
Mr @andriusch created a follow up PR: https://github.com/toptal/granite/pull/77

All good 😄

Reply:
> most of the changes aren't advertising for context

well, let's disagree 🤣 

![image](https://user-images.githubusercontent.com/65587/185381645-63ddeb06-0b38-435c-80e7-b18b407448bb.png)